### DISCUSSION TITLE
Fragen rund um "unignore" - speziell für Parameter aus dem MASTER-Paramset

### DISCUSSION THREAD

**Discussion by Baxxy13** [SukramJ/aiohomematic#1397] [Ideas]
Nachdem ich ja gestern echt überrascht über die "unignore"-Fähigkeiten (bezüglich MASTER-Parameter) war habe ich mich mal drangesetzt und ein paar Tests gemacht. Dabei bin ich auf Probleme gestoßen und daraus haben sich ein paar Fragen ergeben.
Ich hatte mich dann nochmal intensiv mit der Doku und dem verlinkten Code befasst, muss aber auch zugeben das ich den Code nur "so halb" verstehe.

1. Gibt es MASTER-Parameter die sich auch nicht mittels "unignore" integrieren lassen?
(ich hatte diverse MASTER-Parameter für HM/HmIP getestet, von denen einige gingen und andere nicht) 

2. Warum gibt es unterschiedliche Behandlungen der Parameter? (abhängig vom Kanal?)
Beispiel  HmIP / Kanal:0 / GLOBAL_BUTTON_LOCK
geht: 
`GLOBAL_BUTTON_LOCK@HmIP-eTRV-B1:0:MASTER`
geht nicht:
`MASTER:GLOBAL_BUTTON_LOCK`
`GLOBAL_BUTTON_LOCK`

   Beispiel HmIP / Kanal:1 / TEMPERATURE_MAXIMUM
   geht:
   `TEMPERATURE_MAXIMUM@HmIP-STHD:1:MASTER`
   `MASTER:TEMPERATURE_MAXIMUM`
   geht nicht:
   `TEMPERATURE_MAXIMUM`

3. Bekommst du das auch hin das "unignore" für HM Geräte (MASTER_PARAMETER) funktioniert?
Beispiel HM / Gerät (HM-TC-IT-WM-W-EU) / GLOBAL_BUTTON_LOCK
geht nicht:
`GLOBAL_BUTTON_LOCK@HM-TC-IT-WM-W-EU:MASTER`
`MASTER:GLOBAL_BUTTON_LOCK`
`GLOBAL_BUTTON_LOCK`

Hier ist ja das Problem das die MASTER-Parameter nicht nur auf einem Kanal, sondern (wie hier im Beispiel) direkt im Gerät liegen können.

**Comment by SukramJ** [comment]
Danke fürs testen.

zu 1.: nicht bewusst
zu 2.: `GLOBAL_BUTTON_LOCK` wenn kein Paramset angegeben wird wird `VALUES` verwendet.
`MASTER:GLOBAL_BUTTON_LOCK` sollte gehen, aber bestimmt ein Problem mit dem nicht explizit angegebenen Kanalwert 0. Muss ich mir anschauen.
zu 3. Es wird immer eine Kanalnummer vorausgesetzt. Da ich nur noch HmIP Geräte habe denke ich immer nicht an den `ohne Nummer` Kanal. Muss ich mir anschauen.

**Reply by SukramJ** [reply]
Ich tendiere dazu bei `MASTER` nur die Langschreibweise vorzusehen, wobei es dann auch ohne Kanalnummer gegen sollte.
`MASTER:GLOBAL_BUTTON_LOCK` ist damit überflüssig.

**Reply by Baxxy13** [reply]
Danke für's Feedback. 
Ich habe auch schon über "Optimierungen" nachgedacht, die natürlich wieder Arbeit für Dich bedeuten würden. 😉 
Schreibe ich nachher was zu, also bitte noch warten mit dem umcoden.

**Reply by SukramJ** [reply]
Mein Problem beim  MASTER paramset von HM-Geräten ist, das bei allen Parametern OPERATIONS == 0 ist, und ich nicht genau weiss, was ich damit anfangen soll. Event macht ja keinen Sinn, aber Read & Write sollte ja gehen. Oder?

**Reply by Baxxy13** [reply]
Ja read/write per put/get Paramset (XML-RPC) geht. Oder machst du das über die JSON-Api? Müsste ja auch gehen.
Events gibt's da nicht glaube ich.

**Reply by SukramJ** [reply]
XmlRPC. 
JsonRPC verwende ich nur für die zusätzlichen Metadaten und Skripte.

**Reply by Baxxy13** [reply]
Ah, ok.
Das geht, mache ich ab und zu per Dienst "Homematic(IP) Local: Parametersatz schreiben"

```yaml
service: homematicip_local.put_paramset
data:
  device_id: 17e2ec4b65c40e44152609aaa4ef6ac6
  paramset_key: MASTER
  paramset:
    MODUS_BUTTON_LOCK: true
```
(Modus-Bediensperre beim HM-Wandthermostat)

**Comment by Baxxy13** [comment]
Hier nun mal mein Vorschlag.

Die "unignore" - Zeilen vereinheitlichen, nach dem Schema:
`PARAMETER:PARAMSET@DEVICE`

Beispiele:
MASTER:
`GLOBAL_BUTTON_LOCK:MASTER@all` --> alle Geräte die den Master-Parameter haben
`GLOBAL_BUTTON_LOCK:MASTER@HmIP-eTRV-B1` --> nur eTRV-B1

VALUES:
`FROST_PROTECTION:VALUES@all` --> alle Geräte die den Values-Parameter haben
`FROST_PROTECTION:VALUES@HmIP-BWTH` --> nur BWTH

Hätte m.E. den Vorteil das die Anzahl der Elemente (3) samt Separatoren immer gleich ist und du die Prüfung auf Korrektheit vereinfachen kannst. Außerdem macht es "unignore" übersichtlicher.
Den ganz einfachen  `FROST_PROTECTION` würde ich wegrationalisieren da der ja im 1. VALUES-Beispiel integriert ist.



#########################################################################
Ab hier dann eher optional oder vielleicht nice-to-have...

Ob eine Begrenzung auf Kanäle nötig ist, darüber bin ich mir aktuell unsicher. Die meisten Parameter existieren ja nur 1x pro Gerät, aber wenn wir mal als Beispiel `PROCESS` bei einer HmIP-PSM nehmen dann gibt es den 4x.
Hat mich selbst bisher nicht gestört, da ich die nicht benötigten `PROCESS` einfach in HA inaktiv habe. 

Die Beispiele inklusive Kanal sähen dann so aus:
MASTER:
`GLOBAL_BUTTON_LOCK:MASTER@all:allChn` --> alle Geräte die den Master-Parameter, egal wo, haben
`GLOBAL_BUTTON_LOCK:MASTER@all:0` --> alle Geräte die den Master-Parameter auf Kanal:0 haben
`GLOBAL_BUTTON_LOCK:MASTER@HmIP-eTRV-B1:0` --> nur eTRV-B1, Master-Parameter nur von Kanal:0

VALUES:
`PROCESS:VALUES@all:allChn` --> alle Geräte die den Values-Parameter, egal wo, haben
`PROCESS:VALUES@all:3` --> alle Geräte die den Values-Parameter auf Kanal:3 haben
`PROCESS:VALUES@HMIP-PSM:3` --> nur PSM, Values-Parameter nur von Kanal:3

**Reply by SukramJ** [reply]
Ist auf jeden Fall ein Breaking Change.
Größere Umbauten möchte ich eigentlich nicht vornehmen, aber ich behalte das mal im Hinterkopf.

**Reply by Baxxy13** [reply]
Alles klar.
Das das ein Breaking Change wäre habe ich nicht bedacht. 
Der Aufwand für den User wäre m.E. auch nicht allzu groß.
Einmalig die "unignore" anpassen und HA neu starten.

Schwer zu sagen wie intensiv "unignore" genutzt wird.
Bei mir wär's aktuell (für die Variante ohne Kanal):
alt:
```
DECISION_VALUE
INHIBIT
WORKING
PROCESS
DIRECTION
ACTIVITY_STATE
ACTUAL_TEMPERATURE
SET_POINT_TEMPERATURE
SET_TEMPERATURE
ACTIVE_PROFILE
LEVEL@HmIP-BROLL:3:VALUES
LEVEL@HmIP-BROLL:4:VALUES
```

neu: 
```
DECISION_VALUE:VALUES@all:allChn
INHIBIT:VALUES@all:allChn
WORKING:VALUES@all:allChn
PROCESS:VALUES@all:allChn
DIRECTION:VALUES@all:allChn
ACTIVITY_STATE:VALUES@all:allChn
ACTUAL_TEMPERATURE:VALUES@all:allChn
SET_POINT_TEMPERATURE:VALUES@all:allChn
SET_TEMPERATURE:VALUES@all:allChn
ACTIVE_PROFILE:VALUES@all:allChn
LEVEL:VALUES@HmIP-BROLL
```

**Reply by SukramJ** [reply]
Womit ich noch hadere ist, das bei MASTER auch allChn möglich sein soll.
Grundsätzlich werden alle VALUES ParamsetDescriptions geladen, und MASTER nur selektiv, da die MASTER ParamsetDescriptions wesentlich grösser sind.

**Reply by SukramJ** [reply]
Und bei MASTER sollte auch immer das Gerät spezifiziert werden. Gleicher Grund wie oben.

**Reply by SukramJ** [reply]
Du könntest mal bitte die 1.55.0b0 testen.
Bei MASTER müssen Kanal und Gerät immer mit angegeben werden.
Der Platzhalter für Gerät und Kanal ist beides mal `all`.

Könntest Du dich auch um die Aktualisierung der [unignore Beschreibung](https://github.com/danielperna84/hahomematic/blob/devel/docs/unignore.md) kümmern?

**Reply by Baxxy13** [reply]
Mache ich gerne. Wird aber erst frühestens Dienstag was, denke ich.

**Comment by Baxxy13** [comment]
So die Beta 1.55.0b1 ist drauf.
Erste Tests habe ich auch gemacht mit gemischten Ergebnissen.

Fangen wir mal bei HM an (HM-ES-PMSw1-Pl):
**Fall-A1:** `PARAMETER` (aus VALUES) 
```
DECISION_VALUE
INHIBIT
WORKING
```
geht weiterhin wie gehabt (VALUES verfügbar)

**Fall-B1:** `PARAMETER:VALUES@all:all`
```
DECISION_VALUE:VALUES@all:all
INHIBIT:VALUES@all:all
WORKING:VALUES@all:all
```
VALUES verfügbar

**Fall-C1:** `PARAMETER:VALUES@DEVICE:all`
```
DECISION_VALUE:VALUES@HM-ES-PMSw1-Pl:all
INHIBIT:VALUES@HM-ES-PMSw1-Pl:all
WORKING:VALUES@HM-ES-PMSw1-Pl:all
```
VALUES verfügbar

**Fall-D1:** `PARAMETER:VALUES@DEVICE:KANAL`
```
DECISION_VALUE:VALUES@HM-ES-PMSw1-Pl:3
INHIBIT:VALUES@HM-ES-PMSw1-Pl:1
WORKING:VALUES@HM-ES-PMSw1-Pl:1
```
VALUES verfügbar

Bis hierhin mit Parametern aus VALUES alles i.O.
###########################################################
Jetzt werden die Testszenarien um einen MASTER-Parameter erweitert.

**Fall-A2:** `PARAMETER` (aus VALUES) + `PARAMETER:MASTER@DEVICE:KANAL`
```
DECISION_VALUE
INHIBIT
WORKING
AVERAGING:MASTER@HM-ES-PMSw1-Pl:2
```
VALUES verfügbar, MASTER nicht verfügbar

**Fall-B2:** `PARAMETER:VALUES@all:all` + `PARAMETER:MASTER@DEVICE:KANAL`
```
DECISION_VALUE:VALUES@all:all
INHIBIT:VALUES@all:all
WORKING:VALUES@all:all
AVERAGING:MASTER@HM-ES-PMSw1-Pl:2
```
VALUES und MASTER nicht verfügbar

**Fall-C2:** `PARAMETER:VALUES@DEVICE:all` + `PARAMETER:MASTER@DEVICE:KANAL`
```
DECISION_VALUE:VALUES@HM-ES-PMSw1-Pl:all
INHIBIT:VALUES@HM-ES-PMSw1-Pl:all
WORKING:VALUES@HM-ES-PMSw1-Pl:all
AVERAGING:MASTER@HM-ES-PMSw1-Pl:2
```
VALUES verfügbar, MASTER nicht verfügbar

**Fall-D2:** `PARAMETER:VALUES@DEVICE:KANAL` + `PARAMETER:MASTER@DEVICE:KANAL`
```
DECISION_VALUE:VALUES@HM-ES-PMSw1-Pl:3
INHIBIT:VALUES@HM-ES-PMSw1-Pl:1
WORKING:VALUES@HM-ES-PMSw1-Pl:1
AVERAGING:MASTER@HM-ES-PMSw1-Pl:2
```
VALUES verfügbar (korrekt nur für den gewählten Kanal), MASTER nicht verfügbar


**Fragen die sich aus den Test's ergeben:**
- Hattest du jetzt für HM schon was mit implementiert?
- Wenn ja, warum wird mein gewählter MASTER-Parameter `AVERAGING` nicht angelegt?
- Wenn ja, wie ist mit den MASTER-Parametern im Gerät zu verfahren (kein Kanal)

- Warum werden bei B2 die VALUES deaktiviert?

 Wenn ich morgen Zeit habe gucke ich mir mal ein IP-Gerät an.

**Reply by SukramJ** [reply]
AVERAGING hat bei OPERATIONS den Wert 3 (Write, Event). Bisher wird nur 0 durch 6 ersetzt.

**Reply by SukramJ** [reply]
3 ist OK Read, Write

**Reply by Baxxy13** [reply]
Hab mal ein bisschen quer Beet bei HM geguckt, da haben wohl alle Master-Parameter OPERATIONS 3, also Read/Write.

**Reply by SukramJ** [reply]
Danke. Ich schau mir das nochmal an. Das mit dem `all`ist nicht ganz so einfach.

> Wenn ja, wie ist mit den MASTER-Parametern im Gerät zu verfahren (kein Kanal)


GLOBAL_BUTTON_LOCK:MASTER@HM-TC-IT-WM-W-EU:

**Reply by Baxxy13** [reply]
Ich bin da für alles offen wenn das mit dem `all` nicht geht.
Mir gings ja primär um Vereinheitlichung und das man idealerweise in der `unignore` sieht ob man nun einen VALUES oder MASTER Parameter nutzt.

> GLOBAL_BUTTON_LOCK:MASTER@HM-TC-IT-WM-W-EU:

Geht noch nicht. Hängt das auch mit den OPERATIONS zusammen? Wäre hier auch 3.

**Reply by SukramJ** [reply]
Bei mir geht das so. unignore hat in der  b1 noch das Problem, wenn ein Eintrag mit `all` in der Liste ist.

**Reply by Baxxy13** [reply]
Edit: Jetzt geht's, musste den Cache über den Service löschen.

Hmm, zum testen habe ich einzig `GLOBAL_BUTTON_LOCK:MASTER@HM-TC-IT-WM-W-EU:` in der `unignore`.
![HA_Unignore_Single_Master_Param](https://github.com/danielperna84/hahomematic/assets/54898249/4e3e60b9-5a87-44d4-bc9a-4570834ef586)

Das geht hier weiterhin nicht.
![TC_IT_Button_Lock_Unignore](https://github.com/danielperna84/hahomematic/assets/54898249/74f83cf0-62b2-470e-8f18-a7986028cea2)

**Reply by SukramJ** [reply]
Lösch doch mal bitte den Cache.

Teste doch mal mit der neusten hahomematic Version. Ich möchte aktuell kein Release erstellen.

Unter `config/custom_components/homematicip_local` ersetze in der `manifest.json` 
`"requirements": ["hahomematic==2024.1.10"],` durch  `"requirements": ["hahomematic==2024.1.11"],`

- `DECISION_VALUE:VALUES@all:all `wird gekürzt zu `DECISION_VALUE`
- unignore Listen mit `all` in einer Zeile sollten jetzt funktionieren

**Reply by SukramJ** [reply]
GLOBAL_BUTTON_LOCK:MASTER@HM-TC-IT-WM-W-EU:

![20C4F207-7791-4271-A17D-3F370D844D5E](https://github.com/danielperna84/hahomematic/assets/5650106/8369f555-0e45-4d31-a961-d609f014bc62)

**Reply by Baxxy13** [reply]
Cache löschen hat's gebracht. 
Den Rest gucke ich mir später an.

**Reply by Baxxy13** [reply]
Kurzer Zwischenbericht. 
Aktuell gibt's zuviel des Guten.

In `unignore` eingetragen: `AVERAGING:MASTER@HM-ES-PMSw1-Pl:2`
(sonst nichts weiter eingetragen)

Ergebnis:
6 von 7 MASTER-Parameter sichtbar
![HA_Unignore_Single_Master_Param_2](https://github.com/danielperna84/hahomematic/assets/54898249/4e56e56e-a5c3-42c4-8b37-7b8f9da00439)

**Reply by SukramJ** [reply]
Sind die Werte aus dem Averaging Kanal?

**Reply by Baxxy13** [reply]
Ja.
![HA_Unignore_Single_Master_Param_2_SDV](https://github.com/danielperna84/hahomematic/assets/54898249/77ca95a6-312e-4eca-8447-b185fa470b90)

**Reply by SukramJ** [reply]
Ich denke das hat was mit dem Überschreiben der `OPERATIONS` zu tun. 
Bitte lösche den Cache und starte neu.

**Reply by SukramJ** [reply]
Versuch mal bitte die 2024.2.0. Da sollten die überflüssigen Entitäten nicht mehr dabei sein.

**Reply by SukramJ** [reply]
Mit der aktuellen HA Beta lässt sich auch die 1.50.0b2 installieren.

**Comment by Baxxy13** [comment]
Nächste Runde, das Testsystem läuft mit der Beta: 1.50.0b2, nach jeder Änderung an der `unignore` wurde der Cache gelöscht und HA neu gestartet.
Sieht schon sehr gut aus, habe aber wieder ein paar Probleme gefunden.
Hier erstmal die `unignore`, ich teste mit HmIP-PSM und HM-ES-PMSw1-Pl die ja sehr ähnlich sind.
```
LOCAL_RESET_DISABLED:MASTER@HMIP-PSM:0
LOCAL_RESET_DISABLE:MASTER@HM-ES-PMSw1-Pl:
COND_TX_THRESHOLD_HI:MASTER:MASTER@HMIP-PSM:7
COND_TX_THRESHOLD_LO:MASTER:MASTER@HMIP-PSM:7
COND_TX_THRESHOLD_HI_POWER:MASTER@HM-ES-PMSw1-Pl:3
COND_TX_THRESHOLD_LO_POWER:MASTER@HM-ES-PMSw1-Pl:3
AVERAGING:MASTER@HMIP-PSM:6
AVERAGING:MASTER@HM-ES-PMSw1-Pl:2
```

A: 
```
LOCAL_RESET_DISABLED:MASTER@HMIP-PSM:0
LOCAL_RESET_DISABLE:MASTER@HM-ES-PMSw1-Pl:
```
Typ: BOOL; Funktioniert bei HmIP korrekt, bei HM kann ich zwar "einschalten" (und das wird auch so im Gerät gesetzt) aber der Schalter geht gleich wieder "aus" und man hat keine Möglichkeit aus der HA-UI heraus das wieder zu "deaktivieren". 

B:
```
COND_TX_THRESHOLD_HI:MASTER:MASTER@HMIP-PSM:7
COND_TX_THRESHOLD_LO:MASTER:MASTER@HMIP-PSM:7
COND_TX_THRESHOLD_HI_POWER:MASTER@HM-ES-PMSw1-Pl:3
COND_TX_THRESHOLD_LO_POWER:MASTER@HM-ES-PMSw1-Pl:3
```
Typ: bei HM korrekt FLOAT, bei IP falsch INTEGER
Funktioniert bei HM korrekt, wird mit Einheit angezeigt und lässt sich auch setzen.
Wird bei IP nicht angelegt. Schuld dürfte eQ-3 sein da die Description völlig daneben ist.
`(control=, default=10.000, flags=1, id=COND_TX_THRESHOLD_HI, maximum=16.777.215, minimum=0, operations=3, tabOrder=0, type=INTEGER, unit=)`

C:
```
AVERAGING:MASTER@HMIP-PSM:6
AVERAGING:MASTER@HM-ES-PMSw1-Pl:2
```
Typ: INTEGER
Wird bei HM und IP angelegt, bei IP mit vermurkster Einheit (dank eQ-3).
Änderungen werden aber nicht ausgeführt bzw. ins Gerät geschrieben.


Gedanken:
A: Das Problem bei HM ist natürlich ungünstig, muss ich bei Gelegenheit mal testen ob das alle "Switches" betrifft. Vielleicht hast du ja dafür eine Lösung.

B: Vermutlich wird das für IP nicht angelegt weil es keine Unit gibt ober allgemein die Description unstimmig ist. Zwar nicht schön aber dann ist das halt so.

C: 
Hier fällt mir nichts zu ein. Scheint als ob der Service-Call nicht (korrekt) rausgeht.

**Reply by SukramJ** [reply]
zu B
COND_TX_THRESHOLD_HI/COND_TX_THRESHOLD_LO ist Integer HMIP-PSM also richtig.
In pydevccu hat der HM-ES-PMSw1-Pl  kein COND_TX_THRESHOLD_HI_POWER oder COND_TX_THRESHOLD_LO_POWER

Grundsätzlich gilt aber, das sobald ein Parameter per `unignore` sichtbar gemacht wird, das die Standardregeln wie für "normale" Parameter gelten bzw. kein Unterschied zu diesen mehr da ist.

Das Schreiben/Aktualisieren bei MASTER Parametern ist allerdings ein ganz anderer Prozess als bei VALUES Parametern.
Ein Wert wird gesendet -> CONFIG_PENDING ist `wahr`.
Wenn der Wert auf das Gerät übertragen wurde, dass ist CONFIG_PENDING wieder `falsch`, und die Integration liest die aktuellen MASTER Parameter Werte für das Gerät/Kanal. Wenn der Vorgang zu lange dauert, dann setzt HA den eigenen Wert wieder auf den Ursprungswert zurück.

Das ist sicher ein Punkt für die Doku. Eine Änderung eines MASTER Parameter ist eine Konfigurationsänderung, die auch ins Gerät geschrieben werden muss.

**Reply by Baxxy13** [reply]
zu B: Integer bei der PSM ist ok, nur warum werden die beiden nicht angelegt? Wegen der fehlenden Einheit?

> In pydevccu hat der HM-ES-PMSw1-Pl kein COND_TX_THRESHOLD_HI_POWER oder COND_TX_THRESHOLD_LO_POWER

Ok, dann hier mal die Device-Definition. 
[HM-ES-PMSw1-Pl.json](https://github.com/danielperna84/hahomematic/files/14130788/HM-ES-PMSw1-Pl.json)

> Wenn der Vorgang zu lange dauert,

Mag sein das die Config's bei HM und IP unterschiedlich lange zum Übertragen brauchen, das ganze geht aber hier laut Analyzer bei HM und bei IP in unter einer Sekunde. Der Schalter geht nach ca. 2-3s zurück auf AUS obwohl der Wert im Gerät "EIN" ist.

Das die MASTER-Parameter in den Flash-ROM der Geräte geschrieben werden wird natürlich dann auch in der Doku erwähnt.

**Reply by SukramJ** [reply]
> nur warum werden die beiden nicht angelegt? 
COND_TX_THRESHOLD_HI:MASTER:MASTER@HMIP-PSM:7
COND_TX_THRESHOLD_LO:MASTER:MASTER@HMIP-PSM:7

MASTER:MASTER geht nicht.

**Reply by SukramJ** [reply]
Das Problem beim Schreiben existiert nur bei HM?

**Reply by SukramJ** [reply]
> Ok, dann hier mal die Device-Definition.
[HM-ES-PMSw1-Pl.json](https://github.com/danielperna84/hahomematic/files/14130788/HM-ES-PMSw1-Pl.json)

Ich brauch immer beide Beschreibungen.

**Reply by Baxxy13** [reply]
Oh man... ich muss mal die Brille putzen, ist ja peinlich. 
`COND_TX_THRESHOLD_HI:MASTER@HMIP-PSM:7` wird angelegt, schreiben geht hier nicht.  
Mit dem Dienst klappt es (IP):
```yaml
service: homematicip_local.put_paramset
data:
  device_id: 26add84bfe6ae61442fdbcc39a6d4d27
  channel: 7
  paramset_key: MASTER
  paramset:
    COND_TX_THRESHOLD_HI: 5000
```
Bei HM geht schreiben auf (`COND_TX_THRESHOLD_HI_POWER:MASTER@HM-ES-PMSw1-Pl:3`) nun doch. Hatte hier irgendwas versemmelt. Wird Zeit für ne Pause.

Das mit der korrekten Rückmeldung bei BOOL klappt bei IP, bei HM nicht. Ist halt blöd wenn ich den (`GLOBAL_BUTTON_LOCK:MASTER@HM-TC-IT-WM-W-EU:`) nutze um den Lock zu aktivieren, ich ihn dann aber nur per Service oder über die WebUI wieder deaktivieren kann.

Fehlende Datei:
[HM-ES-PMSw1-Pl.json](https://github.com/danielperna84/hahomematic/files/14131162/HM-ES-PMSw1-Pl.json)

**Reply by SukramJ** [reply]
Ich bin mir noch nicht sicher, warum der PSM Wert in HA als Float angezeigt wird, obwohl die Integration das richtig als Integer anlegt.
Schau ich mir morgen mal an.

**Reply by Baxxy13** [reply]
Das mit dem Float / Integer wird auch das Problem bei AVERAGING sein. Ist ja bei HM und IP INTEGER und wird als FLOAT angelegt.
Das schreiben klappt nicht, mit dem Dienst und übergebenem INTEGER hingegen schon.

**Reply by SukramJ** [reply]
Das sollte morgen kein Problem mehr sein.

**Reply by Baxxy13** [reply]
Wir nähern uns dem Finale. 
ENUMS funktionieren auch, getestet mit `HEATING_COOLING:MASTER@HmIP-STHD:5`.
Jetzt ist aber Feierabend... 😉

**Reply by SukramJ** [reply]
HA liefert bei `number` Entitäten immer `float`. Das war also ein bestehender Bug der mit der 1.55.0b3 behoben wurde.

**Reply by Baxxy13** [reply]
Sehr gut, das läuft jetzt auch.

Bleibt erstmal nur das:
`MODUS_BUTTON_LOCK:MASTER@HM-TC-IT-WM-W-EU:`
![GIF 02 02 2024 13-44-45](https://github.com/danielperna84/hahomematic/assets/54898249/96e41cb2-6581-4632-a8f2-d7a7fee27e29)

Ich habe das HM-TC-IT-WM-W-EU direkt vor mir liegen. 
Sobald ich den `MODUS_BUTTON_LOCK` aktiviere erscheint auf dem Display das Schloss. Das funktioniert also soweit.
Nur das der "Schalter" zurückspringt und damit nicht mehr dem realen Status des `MODUS_BUTTON_LOCK` vom Thermostat entspricht.
"Unlocken" kann ich nur per Dienst oder über die Geräteeinstellungen (WebUI).

Lasse ich alles so (Ende des Gif's) und starte HA neu dann ist der `MODUS_BUTTON_LOCK` korrekt "EIN" und lässt sich auch "AUSschalten".
Aber auch hier springt er zurück auf "EIN".

**Reply by SukramJ** [reply]
> Ein Wert wird gesendet -> CONFIG_PENDING ist wahr.
Wenn der Wert auf das Gerät übertragen wurde, dass ist CONFIG_PENDING wieder falsch, und die Integration liest die aktuellen MASTER Parameter Werte für das Gerät/Kanal. Wenn der Vorgang zu lange dauert, dann setzt HA den eigenen Wert wieder auf den Ursprungswert zurück.

Kann es sein das HM-Geräte bei Änderungen von MASTER-Parametern kein CONFIG_PENDING Event senden?

**Reply by Baxxy13** [reply]
Gute Frage, die ich aktuell nicht beantworten kann. 
Muss ich mal schauen ob ich da was herausfinden kann. Aber erstmal Wochenende.

**Reply by SukramJ** [reply]
Also ich sehe bei HM kein `CONFIG_PENDING` Event zurückkommen wie bei HmIP.
Für HM bleibt somit nur die Möglichkeit zu pollen. HA sieht dafür den Service `homeassistant.update_entity` vor

**Reply by SukramJ** [reply]
Konntest Du schon was für die HM Master Parameter rausfinden. Mein Testgerät ist da leider nicht besonders aussagekräftig.

**Reply by Baxxy13** [reply]
Entschuldige, ich hatte viel um die Ohren und bin irgendwie vom Thema abgekommen. 

Habe das nun aber mit 2 verschiedenen HM Geräten getestet.
- A: Batteriegerät HM-Sec-SCo
- B: Netzgerät HM-ES-PMSw1-Pl

**A:**
`SABOTAGE_MSG:MASTER@HM-Sec-SCo:`
Beim "EINschalten" in HA geht der "Schalter" auch erstmal wieder "AUS".
Es wird ein `CONFIG_PENDING = true` - Event erzeugt. (mit dem Homematic-Manager beobachtet)
Sorge ich nun dafür das die Config übertragen wird gibt's ein `CONFIG_PENDING = false` - Event und der "Schalter in HA" springt auf den korrekten Wert. 

**B:** 
**B1:**
`COND_TX_THRESHOLD_HI_POWER:MASTER@HM-ES-PMSw1-Pl:3`
Gebe ich hier einen neuen Wert in HA ein (bspw. 77,7) wird der sauber in das Gerät geschrieben und verbleibt auch so in der HA-UI.
Es ist aber kein `CONFIG_PENDING` - Event im Homematic-Manager zu sehen.

**B2:**
`COND_TX_CYCLIC_ABOVE:MASTER@HM-ES-PMSw1-Pl:3`
Beim "EINschalten" in HA geht der "Schalter" wieder "AUS".
Es ist kein `CONFIG_PENDING` - Event im Homematic-Manager zu sehen.
Damit kann ich bestätigen das zumindest "Always Listener" kein `CONFIG_PENDING` - Event erzeugen. 

Forcieren kann ich das wenn das Gerät zum Zeitpunkt der Config-Änderung nicht erreichbar (aber noch nicht UNREACH) ist. 
Dann wird das in HA zwar "nicht verfügbar", nachdem es aber wieder "da" ist wird die Config übertragen und auch korrekt angezeigt.

**Summarum:**
Leider unterschiedliche Ergebnisse je nach Parameter und Geräteklasse. 
Mir fällt da jetzt auch nichts ein was man machen könnte, außer wie du schon schriebst `homeassistant.update_entity` zu nutzen.

Insgesamt ist die Überarbeitung von `unignore` sehr gut geworden, danke dafür.
Da du inzwischen selbst schon die Doku aktualisiert hast (sorry nochmal) habe ich mir überlegt rund um `unignore` einen Wiki-Eintrag zu erstellen. So mit Step-by-Step Anleitung samt Bildern und Beispielen.

**Reply by SukramJ** [reply]
Dann lass ich das erst mal so.

Bei der Doku hab ich nur das Minimum gemacht. Da kannst Du gerne noch ergänzen, aber ein Wiki-Eintrag hört sich gut an.

**Comment by sebastianlorenzen** [comment]
Auf meinen Fhem Server nutze ich diese integration: https://github.com/zapccu/HMCCU evtl. kann man da gucken wie das umgesetzt ist? Da muss man die Register explizit Pollen nachdem man sie gesetzt hat.

**Comment by dagobert** [comment]
Hi,
Perfect timing for these changes. I was just looking for "missing" entities/capabilities of a device. Luckily you changes just made it to my HA installations.
Somehow I seem to do something wrong. Maybe you can help me.
I am running  HA 2024.2.2
Homematicip_local 1.56

I added the `unignore` file in "/homeassistant/homematicip_local/unignore" (seen in File Editor Add-on).

```
ACOUSTIC_NOTIFICATION_SELECTION:VALUES@HmIP-WRCD:3
DISPLAY_DATA_COMMIT:VALUES@HmIP-WRCD:3
DISPLAY_DATA_STRING:VALUES@HmIP-WRCD:3
DISPLAY_DATA_ID:VALUES@HmIP-WRCD:3
DISPLAY_DATA_BACKGROUND_COLOR:VALUES@HmIP-WRCD:3
DISPLAY_DATA_ICON:VALUES@HmIP-WRCD:3
DISPLAY_DATA_ALIGNMENT:VALUES@HmIP-WRCD:3
DISPLAY_DATA_TEXT_COLOR:VALUES@HmIP-WRCD:3
```

I cleared the cache and restartet HA. No addiitional entities present in Device view. I even deleted the device and reloaded the "ccu-connection". -> nothing

What am I doing wrong?

**Reply by SukramJ** [reply]
> What am I doing wrong?

You need to check the [technical documentation](https://homematic-ip.com/sites/default/files/downloads/hmip-device-documentation.pdf) or [pydevccu](https://github.com/danielperna84/pydevccu/blob/master/pydevccu/paramset_descriptions/HmIP-WRCD.json) (easier).

These parameters are write only (`"OPERATIONS": 2`) so the can't be displayed in HA.
`OPERATIONS` must include 1 (bitwise, `READ`)

**Reply by SukramJ** [reply]
This [discussion](https://github.com/danielperna84/hahomematic/discussions/176) could be useful for HmIP-WRCD.

**Comment by dagobert** [comment]
> These parameters are write only (`"OPERATIONS": 2`) so the can't be displayed in HA. `OPERATIONS` must include 1 (bitwise, `READ`)

Where does this limitation come from? 
In general such write parameter could be translated into input fields (as input helper) with the possible input constraints. For example a dropdown with the options white, black, red. I understand that changing them would not mean that they are applied but at least they could be used in an automation oder script to send them.
The major benefit would be that one would not have to generate multiple input helper for devices that live outside where they belong - they belong to the device.
Is it possible in theory to change homematic_local to "generate" such input fields for write only parameters?

**Reply by SukramJ** [reply]
Theoretically everything is possible, but in practical terms someone who thinks it makes sense has to program it. I personally don't see this point. With regard to the WRCD, the variant with the combined parameter is sufficient.

Sometimes it is not enough to change individual parameters with Homematic or sequences may have to be adhered to. I haven't tried whether this is the case with the wrcd, but it is definitely the case with the sirens, which also have write-only parameters.
But I would guess that this is also the case with wrcd, so changes to individual parameters are not enough.

Build an input helper for the individual parameters in HA, create a send button and send the values ​​of the input helper to the device with a service call. If you like you could post your solution here in the wiki.

**Reply by dagobert** [reply]
Sorry, there is a missunderstanding. You definetly have to send a combined parameter. I already tried it differently many years ago. 

My intention was to use the existing json description of the write input such as type (bool, select and so on) and the selection options to generate the input fields. These could be used in a script to send the combined parameter. The nice thing about it would be to have them within the device view and not loose input_helper.

Since python is not my strongsuit I have sone difficulties to find the right parts in your project(s) where to start, if it comes to adding this.

**Reply by SukramJ** [reply]
Apart from the fact that this does not belong to the initial discussion, it still makes no sense to me, and it wasn't the first time I thought about it.
